public/pages/MonitorDetails/containers/Triggers/Triggers.js

```diff
@@ -39,12 +63,31 @@ export default class Triggers extends Component {
       tableKey: uuidv4(),
       direction: 'asc',
       selectedItems: [],
+      items: [],
+      triggerConditionsById: {},
     };
 
     this.onDelete = this.onDelete.bind(this);
     this.onEdit = this.onEdit.bind(this);
     this.onSelectionChange = this.onSelectionChange.bind(this);
     this.onTableChange = this.onTableChange.bind(this);
+    this.monitorsById = {};
+  }
+
+  async componentDidMount() {
+    const { monitor } = this.props;
+    const triggers = getUnwrappedTriggers(monitor);
+
+    if (monitor.monitor_type === MONITOR_TYPE.COMPOSITE_LEVEL) {
+      for (const trigger of triggers) {
+        const { condition, id } = trigger;
+        await this.initializeTriggerConditionsWithMonitorNames(condition.script.source, id);
+      }
+
+      this.setState({ items: triggers });
+    } else {
+      this.setState({ items: triggers });
+    }
```

Comment:
Non-blocking nitpick: It looks like `this.setState({ items: triggers });` is called in both the `if/else` blocks; so it could be simplified by just moving it out of the block.